packages/core/src/store/reducer.ts

```diff
@@ -71,9 +117,14 @@ export interface State {
   plugins: Record<PluginType, Plugin<any>>
   documents: Record<string, PluginState>
   focus?: string
+  history?: {
```

Comment:
why optional?

Reply:
Because the `initialState` of history is a State too but (normally) doesn't have a history... The current way I think the types would break if I enforce it, because of an infinite loop, wouldn't it? I did this differently previously like:
```typescript
interface BaseState {
  defaultPlugin: PluginType
  plugins: Record<PluginType, Plugin<any>>
  documents: Record<string, PluginState>
  focus?: string
}
export interface State extends BaseState {
  history: {
    initialState: BaseState
    actions: Undoable[][]
    redoStack: Undoable[][]
  }
}
```
Do you like that more?

What I liked about having `initialState: State` is that you can easily extend it to the logic we discussed with in between saves, like this:

```js
state = {
  // ...current state after 6 actions,
  history: {
    initialState: {
        // ...olderState after 3 actions,
      history: {
        initialState: initial,
        actions: [action1, action2, action3]
      }
    },
    actions: [action4, action5, action6]
  }
}
```

Reply:
Okay, I see the reasoning behind that. Fine by me :)